cmd/cli/serve/util.go

```diff
@@ -167,16 +170,46 @@ func getAllowListedLocalPathsConfig() []string {
 	return viper.GetStringSlice(types.NodeAllowListedLocalPaths)
 }
 
-func getNetworkConfig() (node.NetworkConfig, error) {
+func getNetworkConfig(nodeID string) (node.NetworkConfig, error) {
 	var networkCfg types.NetworkConfig
 	if err := config.ForKey(types.NodeNetwork, &networkCfg); err != nil {
 		return node.NetworkConfig{}, err
 	}
+
+	if networkCfg.AuthSecret == "" {
+		// Generate an auth token using the user's client key. It should not be
+		// possible to compute this value by anyone other than the NATS server, and
+		// should be stable across restarts of the node.
+		var keySig string
+		keySig, err := system.SignForClient([]byte(nodeID))
+		if err != nil {
+			return node.NetworkConfig{}, err
+		}
+		hash := sha256.Sum256([]byte(keySig))
+		networkCfg.AuthSecret = base64.RawURLEncoding.EncodeToString(hash[:])
```

Comment:
The key is persistent, and the algorithm used to derive the token from it is (I'm pretty sure?) deterministic, so as long as the key doesn't change this will derive the same token each time.

Reply:
disregard, hadn't comprehended this was derived from the nodeID->peerID->private_key - which _is_ persisted to disk. 

Something else that crossed my mind: iirc we discussed allowing users to set arbitrary(?) NodeIDs (names?) at some point, and while I understand we aren't there yet, we'll need to be careful with its implementation since changes to it will result in different auth secrets.

regardless lgtm.